lib/pages/chat/empty_chat.dart

```diff
@@ -0,0 +1,454 @@
+      if (result.result != null) {
+        final roomId = result.result;
+        if (onRoomCreatedSuccess != null && roomId != null) {
+          final room = Matrix.of(context).client.getRoomById(roomId);
+          if (room != null) {
+           await onRoomCreatedSuccess(room);
+          }
+          VRouter.of(context).toSegments(
+            ['rooms', 'empty', roomId],
+            isReplacement: true,
+          );
+        }
+      }
+    }).onError((error, stackTrace) {
+      Logs().d("_createRoom: $error");
+      VRouter.of(context).pop();
+    })
+    .timeout(const Duration(minutes: 5));
+  }
+
+  void onFileTapped({required Event event}) async {
+    final permissionHandler = PermissionHandlerService();
+    final storagePermissionStatus = await permissionHandler.storagePermissionStatus;
+    switch (storagePermissionStatus) {
+      case PermissionStatus.denied:
+        await showDialog(
+          useRootNavigator: false,
+          context: context,
+          builder: (context) {
+            return PermissionDialog(
+              permission: Permission.storage,
+              explainTextRequestPermission: Text(L10n.of(context)!.explainStoragePermission),
+              icon: const Icon(Icons.preview_outlined),
+            );
+          }
+        );
+        if (await permissionHandler.storagePermissionStatus == PermissionStatus.granted) {
+          _handleDownloadFileForPreview(event: event);
+        }
+        break;
+      case PermissionStatus.granted:
+        _handleDownloadFileForPreview(event: event);
+        break;
+      case PermissionStatus.permanentlyDenied:
+        showDialog(
+          useRootNavigator: false,
+          context: context,
+          builder: (context) {
+            return PermissionDialog(
+              permission: Permission.storage,
+              explainTextRequestPermission: Text(L10n.of(context)!.explainGoToStorageSetting),
+              icon: const Icon(Icons.preview_outlined),
+            );
+          }
+        );
+        break;
+      case PermissionStatus.restricted:
+      case PermissionStatus.limited:
+      case PermissionStatus.provisional:
+        break;
+    }
+  }
+
+
+  void _handleDownloadFileForPreview({required Event event}) async {
+    final downloadFileForPreviewInteractor = getIt.get<DownloadFileForPreviewInteractor>();
+    final tempDirPath = (await getTemporaryDirectory()).path;
+    downloadFileForPreviewInteractor.execute(
+      event: event,
+      tempDirPath: tempDirPath,
+    ).listen((event) {
+      event.fold(
+        (failure) {
+          if (failure is DownloadFileForPreviewFailure) {
+            Fluttertoast.showToast(msg: 'Error: ${failure.exception}');
+          }
+        },
+        (success) {
+          if (success is DownloadFileForPreviewSuccess) {
+            _openDownloadedFileForPreview(downloadFileForPreviewResponse: success.downloadFileForPreviewResponse);
+            Navigator.of(context).pop();
```

Comment:
IMO, `Navigator.of(context).pop();` should be called before `_openDownloadedFileForPreview` to avoid cases where it closes the open file dialog

Reply:
remove the code, and do it in next ticket